commands/user.bee.inc

```diff
@@ -98,6 +196,204 @@ function users_bee_callback($arguments, $options) {
+    )), 'success');
+  }
+  else {
+    bee_message(bt("Could not create a new user account with the name, !user", array(
+      '!user' => $name,
+    )), 'error');
+  }
+}
+
+/**
+ * Command callback: Delete/cancel a given user.
+ */
+function user_cancel_bee_callback($arguments, $options) {
+  // Load the user account.
+  $user = user_load_by_name($arguments['username']);
+  if (!$user) {
+    bee_message(bt("User account '!user' cound not be found.", array(
+      '!user' => $arguments['username'],
+    )), 'error');
+    return;
+  }
+  $answer = bee_confirm(bt("Are you sure to delete the user !user?", array(
+    '!user' => $arguments['username'],
+    )), FALSE);
+  if (!$answer) {
+    return;
+  } 
+
+  user_cancel(array(
+    'user_cancel_notify' => TRUE,
+    'user_cancel_method' => 'user_cancel_delete',
+  ), $user->uid, 'user_cancel_delete');
+
+  // Invoke the Batch API, disable backdrop_goto(), and execute.
+  $batch = &batch_get();
+  $batch['progressive'] = FALSE;
+  batch_process();
+
+  bee_message(bt("User account '!user' has been removed.", array(
+    '!user' => $user->name,
+  )), 'success');
+}
+
+/**
+ * Command callback: Block a given user.
+ */
+function user_block_bee_callback($arguments, $options) {
+  // Load the user account.
+  $user = user_load_by_name($arguments['username']);
+  if (!$user) {
+    bee_message(bt("User account '!user' cound not be found.", array(
+      '!user' => $arguments['username'],
+    )), 'error');
+    return;
+  }
+
+  $user->status = 0;
+  user_save($user);
+
+  bee_message(bt("User '!user' has been blocked.", array(
+    '!user' => $user->name,
+  )), 'success');
+}
+
+/**
+ * Command callback: Unblock a given user.
+ */
+function user_unblock_bee_callback($arguments, $options) {
+  // Load the user account.
+  $user = user_load_by_name($arguments['username']);
+  if (!$user) {
+    bee_message(bt("User account '!user' cound not be found.", array(
+      '!user' => $arguments['username'],
+    )), 'error');
+    return;
+  }
+
+  $user->status = 1;
+  user_save($user);
+
+  bee_message(bt("User '!user' has been unblocked.", array(
+    '!user' => $user->name,
+  )), 'success');
+}
+
+/**
+ * Command callback: Add a role to a given user.
+ */
+function user_add_role_bee_callback($arguments, $options) {
+  $user = user_load_by_name($arguments['username']);
+  if (!$user) {
+    bee_message(bt("User account '!user' cound not be found.", array(
+      '!user' => $arguments['username'],
+    )), 'error');
+    return;
+  }
+  $roles = user_roles(TRUE);
+  if (empty($roles[$arguments['role']])) {
+    $answer = bee_confirm(bt("The role !role doesn't exists, would you like to create it?", array(
+      '!role' => $arguments['role'],
+      )), FALSE);
+    if (!$answer) {
+      return;
+    } 
+  }
+  $role = new stdClass();
+  $role->name = $arguments['role'];
+  $role->label = $arguments['role'];
+  if (user_role_save($role)) {
```

Comment:
Sorry, I wasn't clear enough here.  I can't see any benefit to creating the role here if it doesn't exist:

- We'd be creating a role where they'd need to go and correct the label. Even if we tried to make the first letter upper case that wouldn't be much benefit if the role name had underscores to represent spaces.  We could tie ourselves in knots trying to second guess what the label should be.
- Roles are not much use without permissions and therefore we shouldn't really start what we can't finish

Therefore, best to just do an error message if the role doesn't exist.  An alternative, but more complex, approach, would be to present a list of valid roles, allow them to select the one they meant to use and then use that.